R/zchunk_L100.Population_downscale_ctry.R

```diff
@@ -33,54 +34,191 @@ module_socioeconomics_L100.Population_downscale_ctry_DISABLED <- function(comman
+      group_by(year) %>%
+      mutate(pop = if_else(is.na(pop), 0, pop)) %>%
+      summarize(pop = sum(pop))
+    # Subtract reported countries from total global - these are the total population values in each period that will be allocated to countries with missing values
+    pop_missing <- left_join(pop_global, pop_notmissing, by = "year") %>%
+      mutate(pop_allocate = pop.x - pop.y) %>%
+      select(year, pop_allocate) %>%
+      mutate(year = paste("X", year, sep = "")) %>%
+      spread(year, pop_allocate)
+
+    # Fifth, estimate historic population values for countries with missing values and calculate ratios relative to 1950
+    maddison_hist_ratio <- filter(hist_interp, iso != "world_total") %>%
+      ungroup() %>%
+      mutate(year = paste("X", year, sep = "")) %>%
+      spread(year, pop) %>%
+      # Fill in missing values for period t using ratio of country to total population in period t+1 times the missing share in period t.
+      # if(is.na(X1900), sum(X1950))
+      mutate(X1900 = if_else(is.na(X1900), ((X1950 / sum(X1950[is.na(X1900)])) * pop_missing$X1900), X1900)) %>%
+      mutate(X1850 = if_else(is.na(X1850), ((X1900 / sum(X1900[is.na(X1850)])) * pop_missing$X1850), X1850)) %>%
+      mutate(X1800 = if_else(is.na(X1800), ((X1850 / sum(X1850[is.na(X1800)])) * pop_missing$X1800), X1800)) %>%
+      mutate(X1750 = if_else(is.na(X1750), ((X1800 / sum(X1800[is.na(X1750)])) * pop_missing$X1750), X1750)) %>%
+      mutate(X1700 = if_else(is.na(X1700), ((X1750 / sum(X1750[is.na(X1700)])) * pop_missing$X1700), X1700)) %>%
+      # Population ratios for historic years compared to 1950 for all countries (will be used to scale UN populations from 1950)
+      mutate(R1900 = X1900 / X1950) %>%
+      mutate(R1850 = X1850 / X1950) %>%
+      mutate(R1800 = X1800 / X1950) %>%
+      mutate(R1750 = X1750 / X1950) %>%
+      mutate(R1700 = X1700 / X1950) %>%
+      select(-X1950, -X1900, -X1850, -X1800, -X1750, -X1700) %>%
+      gather(year, pop_ratio_1950, -iso) %>%
+      mutate(year = as.integer(substr(year, 2, 5)))
+    # Need to create matching iso codes for three countries in UN, but not Maddison.
+    # Set Serbia and Montenegro ratio equal to Serbia & Montenegro and use Indonesia population ratio for East Timor
+    mne_srb_tls <- filter(maddison_hist_ratio, iso %in% c("idn", "scg")) %>%
+      spread(iso, pop_ratio_1950) %>%
+      rename(mne = scg, tls = idn) %>%
+      mutate(srb = mne) %>%
+      gather(iso, pop_ratio_1950, -year)
+    # Combine with other ratio_iso values
+    maddison_hist_ratio <- filter(maddison_hist_ratio, iso != "scg") %>% # Remove Serbia & Montenegro (scg) because we will replace with Serbia (srb) and Montenegro (mne)
+      bind_rows(mne_srb_tls)
+
+    # Sixth, apply Maddison ratios for historic periods to UN populatio data that begin in 1950
+    # Clean raw UN population data
+    un_clean <- UN_popTot %>%
+      filter(Scenario == "EST") %>%  # Historic only (not projections)
```

Comment:
It stands for "estimates". It's only used on the historical years, which is why it's the one scenario we're using here (we don't use the UN for population forecasts)